wandb/sdk/artifacts/artifact.py

```diff
@@ -1055,6 +1080,30 @@ def _update(self) -> None:
                 )
                 .replace("_TTL_DURATION_SECONDS_FIELDS_", "")
             )
+
+        tags_to_add = validate_tags(set(self._tags) - set(self._saved_tags))
+        tags_to_delete = validate_tags(set(self._saved_tags) - set(self._tags))
+        if "tags" in fields:
+            mutation_template = (
+                mutation_template.replace(
+                    "_TAGS_TO_ADD_TYPE_", "$tagsToAdd: [TagInput!],"
+                )
+                .replace("_TAGS_TO_DELETE_TYPE_", "$tagsToDelete: [TagInput!],")
+                .replace("_TAGS_TO_ADD_VALUE_", "tagsToAdd: $tagsToAdd")
+                .replace("_TAGS_TO_DELETE_VALUE_", "tagsToDelete: $tagsToDelete")
```

Comment:
likewise here: why not use a format string instead?

Reply:
(same reason: https://github.com/wandb/wandb/pull/8085#discussion_r1722514453)